contracts/Prover.sol

```diff
@@ -336,6 +337,15 @@ contract Prover is SimpleProver {
         require(
             existingSettlementBlockProof.stateRoot == l1WorldStateRoot, "settlement chain state root not yet proved"
         );
+
+        // check that the End Batch Block timestamp is greater than the current timestamp + finality delay
+        uint256 endBatchBlockTimeStamp = _bytesToUint(RLPReader.readBytes(RLPReader.readList(rlpEncodedBlockData)[11]));
+
+        require(
```

Comment:
y not use custom errors instead of requires?

Reply:
u should test this condition

Reply:
nice catch 🙏 
updated now